src/user/user.controller.ts

```diff
@@ -68,9 +70,31 @@ export class UserController {
     JwtAuthGuard,
     RolesGuard
   )
-  @Patch(':rut')
-  update(@Param('rut') rut: string, @Body() updateUserDto: UpdateUserDto) {
-    return this.userService.update(rut, updateUserDto);
+  @Patch('update')
+  update(@Query('rut') rut: string, @Body() updateUserDto: UpdateUserDto) {
+    try {
+      return this.userService.update(rut, updateUserDto);
+    } catch (error) {
+      throw new BadRequestException("Error updating user")
```

Comment:
UserService already throws an exception if something goes wrong - you don't need to catch it and throw a new one in here